lib/osf-components/addon/components/validated-input/checkboxes/x-checkbox/component.ts

```diff
@@ -10,26 +11,35 @@ import template from './template';
 @tagName('')
 export default class ValidatedInputCheckboxesXCheckbox<T> extends Component {
     // Required arguments
-    relationArray?: T[];
+    relationArray?: DS.PromiseArray<T> | T[];
     option!: T;
     checkboxName!: string;
     ariaLabel!: string;
     disabled!: boolean;
     checkboxId!: string;
 
-    @computed('option', 'relationArray.[]')
+    @computed('option', '_relationArray.[]')
     get checked(): boolean {
-        return Array.isArray(this.relationArray) && this.relationArray.includes(this.option);
+        return this._relationArray.includes(this.option);
     }
 
     set checked(checked: boolean) {
-        if (Array.isArray(this.relationArray)) {
-            if (checked && !this.checked) {
-                this.relationArray.pushObject(this.option);
-            }
-            if (!checked && this.checked) {
-                this.relationArray.removeObject(this.option);
+        if (checked && !this.checked) {
+            this._relationArray.pushObject(this.option);
+        }
+        if (!checked && this.checked) {
+            this._relationArray.removeObject(this.option);
+        }
+    }
+
+    @computed('relationArray.[]')
```

Comment:
Instead of doing this as a computed, does it make more sense to set a `_relationArray` property upon `init` or `didReceiveAttrs`?

Reply:
the computed will recompute if the contents of `relationArray` change, which is nice